### PR TITLE
Add breakout room invitation feature

### LINKED ISSUE #6392: Add Send Breakout Rooms Invitations in HTML5
In flash we have a feature called `Send Breakout Rooms Invitations`, implement this feature in HTML5 client.

### PR DESCRIPTION
**Pull request by Tainan404**
Closes #6392.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx [hunk 1]
```diff
@@ -193,7 +210,17 @@ class ActionsDropdown extends Component {
             label={formatMessage(createBreakoutRoom)}
             description={formatMessage(createBreakoutRoomDesc)}
             key={this.createBreakoutRoomId}
-            onClick={this.handleCreateBreakoutRoomClick}
+            onClick={this.onCreateBreakouts}
+          />
+        )
+        : null),
+      (isUserModerator && !meetingIsBreakout && hasBreakoutRoom && getUsersNotAssigned(users).length
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx]: assign this a variable with a meaningful name `canInviteUsers` or something like that

File: bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx [hunk 2]
```diff
@@ -205,19 +232,28 @@ class ActionsDropdown extends Component {
     mountModal(<PresentationUploaderContainer />);
   }
 
-  handleCreateBreakoutRoomClick() {
+  handleCreateBreakoutRoomClick(invitation) {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx]: `isInvitation`

File: bigbluebutton-html5/imports/ui/components/actions-bar/service.js [hunk 3]
```diff
@@ -40,4 +51,7 @@ export default {
   toggleRecording: () => makeCall('toggleRecording'),
   processOutsideToggleRecording: arg => processOutsideToggleRecording(arg),
   createBreakoutRoom: (numberOfRooms, durationInMinutes, freeJoin = true, record = false) => makeCall('createBreakoutRoom', numberOfRooms, durationInMinutes, freeJoin, record),
+  makeInvitation: (breakoutId, userId) => makeCall('requestJoinURL', { breakoutId, userId }),
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/service.js]: `sendInvitation`

File: bigbluebutton-html5/imports/api/breakouts/server/methods/requestJoinURL.js [hunk 4]
```diff
@@ -12,16 +12,19 @@ export default function requestJoinURL(credentials, { breakoutId }) {
   check(meetingId, String);
   check(requesterUserId, String);
   check(requesterToken, String);
+  const userId = id || requesterUserId;
   const Breakout = Breakouts.findOne({ breakoutId });
-  const BreakoutUser = Breakout.users.filter(user => user.userId === requesterUserId).shift();
+  const BreakoutUser = Breakout.users.filter(user => user.userId === userId).shift();
+
   if (BreakoutUser) return BreakoutUser.redirectToHtml5JoinURL;
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/api/breakouts/server/methods/requestJoinURL.js]: are you reading this value somewhere? otherwise you should not return it.

File: bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx [hunk 5]
```diff
@@ -182,19 +220,29 @@ class ActionsDropdown extends Component {
...
     const {
       createBreakoutRoom,
       mountModal,
       meetingName,
       users,
+      getUsersNotAssigned,
+      getBreakouts,
+      sendInvitation,
     } = this.props;
+    console.error(this.props);
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx]: remove console log

Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx]: Remove console log

File: bigbluebutton-html5/imports/ui/components/actions-bar/service.js [hunk 6]
```diff
@@ -4,6 +4,41 @@ import { makeCall } from '/imports/ui/services/api';
 import Meetings from '/imports/api/meetings';
 import Breakouts from '/imports/api/breakouts';
 
+const processOutsideToggleRecording = (e) => {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/service.js]: Why did you need to add `processOutsideToggleRecording` here? Perhaps slipped during conflict resolving? It was moved to navbar/service.js

Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/actions-bar/service.js]: Same for `connectRecordingObserver`

File: bigbluebutton-html5/imports/api/breakouts/server/methods/requestJoinURL.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ import { check } from 'meteor/check';
 import RedisPubSub from '/imports/startup/server/redis';
 import Breakouts from '/imports/api/breakouts';
 
-export default function requestJoinURL(credentials, { breakoutId }) {
+export default function requestJoinURL(credentials, { breakoutId, userId: id }) {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/breakouts/server/methods/requestJoinURL.js]: Please rename `id` to `userIdToInvite`

File: bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx [hunk 8]
```diff
@@ -128,6 +151,11 @@ class ActionsDropdown extends Component {
       formatMessage,
     } = intl;
 
+    const CanInviteUsers = isUserModerator
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx]: Please use camelCase- `canInviteUsers`

File: bigbluebutton-html5/imports/ui/components/actions-bar/create-breakout-room/component.jsx [hunk 9]
```diff
@@ -77,6 +77,14 @@ const intlMessages = defineMessages({
     id: 'app.audio.backLabel',
     description: 'Back label',
   },
+  invitationTitle: {
+    id: 'app.invitation.title',
+    description: 'isInvitationto breakout title',
+  },
+  invitationConfirm: {
+    id: 'app.invitation.confirm',
+    description: 'isInvitationto breakout cpnfirm button label',
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/create-breakout-room/component.jsx]: `'invitation to breakout confirm button label',`

File: bigbluebutton-html5/imports/ui/components/actions-bar/create-breakout-room/component.jsx [hunk 10]
```diff
@@ -160,15 +175,42 @@ class BreakoutRoom extends Component {
...
+    const breakouts = getBreakouts();
+    if (users.length === this.getUserByRoom(0).length) {
+      this.setState({ valid: false });
+      return;
+    }
+
+    users
+      .filter(u => u.room > 0)
+      .forEach((user) => {
+        const breakout = breakouts[user.room - 1];
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/actions-bar/create-breakout-room/component.jsx]: `breakouts[user.room - 1]` -- is this 100% reliable - could we not filter by an identifier?

**Comment by antobinary** [comment]
Tested - it works well! :clap:
